core/src/main/java/org/elasticsearch/common/transport/TransportAddress.java

```diff
@@ -46,5 +46,9 @@
 
     boolean sameHost(TransportAddress other);
 
+    default boolean isLocalAddress() {
```

Comment:
well, I'm not sure we can assume that all addresses are by default "public". I tend towards saying implementers need to make this call.